SQL/tests/syntax/syntax_test_postgres.psql

```diff
@@ -0,0 +1,311 @@
+--                                            ^^^^^ keyword.other
+--                                                  ^ punctuation.section.group.begin
+--                                                   ^^^^ meta.column-name
+
+
+CREATE TABLE test1 (
+    name                     varchar(500)  NOT NULL PRIMARY KEY,
+    description              varchar       NOT NULL DEFAULT '',
+    authors                  varchar[],
+);
+
+CREATE TABLE test2 (
+    test1name               varchar(500)  NOT NULL REFERENCES test1(name) ON DELETE CASCADE ON UPDATE CASCADE,
+--  ^^^^^^^^^ meta.column-name
+--                          ^^^^^^^^^^^^ storage.type
+--                                        ^^^ keyword.operator.logical
+--                                            ^^^^ constant.language.null
+--                                                 ^^^^^^^^^^ storage.modifier
+--                                                            ^^^^^ meta.table-name
+--                                                                 ^ punctuation.section.group.begin
+--                                                                  ^^^^ meta.column-name
+--                                                                      ^ punctuation.section.group.end
+--                                                                        ^^^^^^^^^^^^^^^^^ storage.modifier
+--                                                                                          ^^^^^^^^^^^^^^^^^ storage.modifier
+--                                                                                                           ^ punctuation.separator.sequence
+    PRIMARY KEY(test1name)
+);
+
+ALTER TABLE test2 ADD something varchar[];
+-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.alter
+-- ^^^^^^^^ keyword.other.ddl
+--          ^^^^^ meta.table-name
+--                ^^^ keyword.other.ddl
+--                    ^^^^^^^^^ meta.column-name
+--                              ^^^^^^^ storage.type
+--                                     ^ punctuation.section.brackets.begin
+--                                      ^ punctuation.section.brackets.end
+--                                       ^ punctuation.terminator.statement
+
+
+CREATE TRIGGER blah AFTER INSERT OR UPDATE
+-- ^^^ keyword.other.ddl
+--     ^^^^^^^ keyword.other
+--             ^^^^ meta.toc-list.full-identifier entity.name.other
+--                  ^^^^^^^^^^^^^^^^^^^^^^ keyword.other
+    ON some_table FOR EACH ROW EXECUTE PROCEDURE some_procedure();
+--  ^^ keyword.other
+--     ^^^^^^^^^^ meta.table-name
+--                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ keyword.other
+--                                               ^^^^^^^^^^^^^^ meta.function-call support.function
+
+
+CREATE FUNCTION highlight_result_array(vals varchar[], something tsquery, something_else boolean) RETURNS varchar[] AS $$
+DECLARE
+-- ^^^^ keyword.declaration.variable
+    output varchar[];
```

Comment:
Is it really only one declaration, which is supported within `DECLARE`?

A second one will break highlighting.